xarray/core/types.py

```diff
@@ -85,6 +89,13 @@ def dtype(self) -> np.dtype:
     Self: Any = None
     DTypeLikeSave: Any = None
 
+if module_available("cftime"):
```

Comment:
This should go into the `if TYPE_CHECKING` branch if you want to keep the cftime import lazy.

Also it seems that this is quite a special case that is not obvious to type correctly.
We need to have an alias for an attribute of an optional package.

I _think_ this requires a `TypeAlias` which itself requires special handling for python versions.

untested:
```python
if sys.version_info >= (3, 10):
    from typing import TypeAlias
else:
    from typing_extensions import TypeAlias

if TYPE_CHECKING:
    try:
        import cftime
        CFTimeDatetime: TypeAlias = cftime.datetime
    except ImportError:
        CFTimeDatetime: Any = None  # type: ignore
```

Reply:
Sorry about that -- I put that code under the `if TYPE_CHECKING` statement now (previously I was being dense regarding the fact that we could also put imports of `CFTimeDatetime` in other modules behind the `if TYPE_CHECKING` guard as well).  

I think we *may* be able to get away without the `TypeAlias` if we are clever in how we import `cftime.datetime` within `types.py`.  See the way I implemented things currently.  Do you think that's an OK approach?

Thanks for your help.